sporkfish/searcher/move_ordering.py

```diff
@@ -69,3 +70,38 @@ def evaluate(self, board: Board, move: chess.Move) -> float:
             ]
         else:
             return 0
+
+
+class KillerMoveHeuristic(MoveOrder):
+    def __init__(self, max_depth: int) -> None:
+        # Initialize a list to store up to two killer moves for each depth
+        # Using a fixed size per depth for simplicity; could be dynamic based on actual usage
+        self.killer_moves: List[List[Optional[chess.Move]]] = [
+            [None, None] for _ in range(max_depth)
+        ]
+
+    def add_killer_move(self, move: chess.Move, depth: int) -> None:
+        """
+        Add a move to the killer moves list for a given depth, ensuring it's not already present.
+        If the move is new, it replaces the oldest killer move.
+
+        Parameters:
+            move (chess.Move): The move to be added as a killer move.
+            depth (int): The depth at which the move proved to be a killer move.
+        """
+        killers = self.killer_moves[depth]
+        if move not in killers:
+            # Shift the existing killer moves down and add the new one at the front
+            killers.pop()
+            killers.insert(0, move)
+
+    def evaluate(self, board: Board, move: chess.Move, depth: int) -> float:
+        killers = self.killer_moves[depth]
+        if move in killers:
```

Comment:
On second thought 

return 1 if move in killers else 0

looks cleaner to me. Sorry!

Reply:
Checking my understanding, so `killer_moves` evaluation simply checks if a move is a [killer](https://rustic-chess.org/search/ordering/killers.html), and gives that move a higher score?

So in fact, most non-killer-moves have very similar scores? How are those being ordered?

Reply:
Killer move heuristic is not great move ordering by itself. It is better when combined with mvvlva. We will redesign the structure when we combine them later. #114

Reply:
worked fine with mypy
```
sporkfish % mypy ./sporkfish/searcher/move_ordering.py
Success: no issues found in 1 source file
```